azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/oauth/AzureAuthorizationCodeFlow.java

```diff
@@ -53,15 +53,35 @@ protected AzureAuthorizationCodeFlow(Builder builder) {
     @Override
     public AuthorizationCodeTokenRequest newTokenRequest(String authorizationCode) {
         AuthorizationCodeTokenRequest request = super.newTokenRequest(authorizationCode);
-        request.setGrantType(DEFAULT_GRANT_TYPE);
+        request.setGrantType(AzureOAuthConstants.DEFAULT_GRANT_TYPE);
         request.setResponseClass(AzureTokenResponse.class);
-        request.put("assertion", authorizationCode);
-        request.put("client_assertion_type", DEFAULT_CLIENT_ASSERTION_TYPE);
-        request.put("client_assertion", clientSecret);
-        request.put("redirect_uri", redirectUri);
+        request.put(AzureOAuthConstants.REQUEST_BODY_FIELD_ASSERTION, authorizationCode);
+        request.put(AzureOAuthConstants.REQUEST_BODY_FIELD_CLIENT_ASSERTION_TYPE, AzureOAuthConstants.DEFAULT_CLIENT_ASSERTION_TYPE);
+        request.put(AzureOAuthConstants.REQUEST_BODY_FIELD_CLIENT_ASSERTION, clientSecret);
+        request.put(AzureOAuthConstants.REQUEST_BODY_FIELD_REDIRECT_URI, redirectUri);
         return request;
     }
 
+    @Override
+    public Credential loadCredential(String userId) throws IOException {
+        Credential credential = super.loadCredential(userId);
+        if (null == credential) {
+            return null;
+        }
+        AzureCredential.Builder credentialBuilder = new AzureCredential.Builder(credential.getMethod());
+        credentialBuilder.setClientAuthentication(credential.getClientAuthentication());
+        credentialBuilder.setClock(credential.getClock());
+        credentialBuilder.setJsonFactory(credential.getJsonFactory());
+        credentialBuilder.setRefreshListeners(credential.getRefreshListeners());
+        credentialBuilder.setRequestInitializer(credential.getRequestInitializer());
+        credentialBuilder.setTokenServerEncodedUrl(credential.getTokenServerEncodedUrl());
+        credentialBuilder.setTransport(credential.getTransport());
+        credentialBuilder.setRedirectUri(redirectUri);
+        credentialBuilder.setClientSecret(clientSecret);
+        credentialBuilder.setCachedRefreshToken(credential.getRefreshToken());
```

Comment:
Can't override getRefreshToken as the method is final.  Since we are using the credential that was stored in the database to create a new credential object, use the getRefreshToken method to save the current refresh token value in the new credential object as a member.